tests/mainnet/metaregistry/integrative/test_getters.py

```diff
@@ -0,0 +1,793 @@
+
+    registry_id, registry_handler, registry, pool = registry_pool_index_iterator[pool_id]
+
+    # we skip if pool was registered: this is because some pools are
+    # registered in the stable registry but are from the stable factory.
+    # skipping does not skip the test: since metaregistry chooses stable
+    # factory as the registry handler and not the registry in these cases
+    if check_pool_already_registered(metaregistry, pool, registry_handler):
+        pytest.skip()
+
+    # we skip if the pool is empty:
+    if sum(metaregistry.get_balances(pool)) == 0:
+        pytest.skip("empty pool: skipping.")
+
+    if brownie.interface.ERC20(metaregistry.get_lp_token(pool)).totalSupply() == 0:
+        pytest.skip("lp token supply is zero.")
+
+    chain.snapshot()
+    registry_reverts = False
+    try:
+        actual_output = _get_admin_balances_actuals(
+            registry_id, registry, pool, metaregistry, alice
+        )
+    except brownie.exceptions.VirtualMachineError:
+        registry_reverts = True
+    chain.revert()
+
+    if registry_reverts:
+        with brownie.reverts():
+            metaregistry.get_admin_balances(pool)
+    else:
+        metaregistry_output = metaregistry.get_admin_balances(pool)
+        for j, output in enumerate(actual_output):
+            try:
+                output == metaregistry_output[j]
+            except AssertionError:
+                diff = output - metaregistry_output[j]
+                assert diff < 100
+                warnings.warn(f"actual_output != metaregistry_output. Diff: {diff}")
+
+
+@pytest.mark.parametrize("pool_id", range(MAX_POOLS))
+def test_get_fees(metaregistry, registry_pool_index_iterator, pool_id):
+
+    skip_if_pool_id_gt_max_pools_in_registry(pool_id, registry_pool_index_iterator)
+
+    registry_id, registry_handler, registry, pool = registry_pool_index_iterator[pool_id]
+
+    # we skip if pool was registered: this is because some pools are
+    # registered in the stable registry but are from the stable factory.
+    # skipping does not skip the test: since metaregistry chooses stable
+    # factory as the registry handler and not the registry in these cases
+    if check_pool_already_registered(metaregistry, pool, registry_handler):
+        pytest.skip()
+
+    # get_fees
+    actuals_reverts = False
+    try:
+        if registry_id != METAREGISTRY_CRYPTO_FACTORY_HANDLER_INDEX:
+            actual_output = registry.get_fees(pool)
+        else:
+            curve_contract = curve_pool_v2(pool)
+            actual_output = [
+                curve_contract.fee(),
+                curve_contract.admin_fee(),
+                curve_contract.mid_fee(),
+                curve_contract.out_fee(),
+            ]
+    except brownie.exceptions.VirtualMachineError:
+        actuals_reverts = True
+
+    if actuals_reverts:
+        with brownie.reverts():
+            metaregistry.get_fees(pool)
```

Comment:
what are the pools for which `get_fees` reverts?
why can't the metaregistry return the right values even if underlying registry reverts?

Reply:
i have now removed all revert handling